crates/storage_impl/src/redis/pub_sub.rs

```diff
@@ -81,122 +97,20 @@ impl PubSubInterface for std::sync::Arc<redis_interface::RedisConnectionPool> {
-                                })
-                                .await;
-                            key
-                        }
-                        CacheKind::All(key) => {
-                            CONFIG_CACHE
-                                .remove(CacheKey {
-                                    key: key.to_string(),
-                                    prefix: self.key_prefix.clone(),
-                                })
-                                .await;
-                            ACCOUNTS_CACHE
-                                .remove(CacheKey {
-                                    key: key.to_string(),
-                                    prefix: self.key_prefix.clone(),
-                                })
-                                .await;
-                            CGRAPH_CACHE
-                                .remove(CacheKey {
-                                    key: key.to_string(),
-                                    prefix: self.key_prefix.clone(),
-                                })
-                                .await;
-                            ROUTING_CACHE
-                                .remove(CacheKey {
-                                    key: key.to_string(),
-                                    prefix: self.key_prefix.clone(),
-                                })
-                                .await;
-                            DECISION_MANAGER_CACHE
-                                .remove(CacheKey {
-                                    key: key.to_string(),
-                                    prefix: self.key_prefix.clone(),
-                                })
-                                .await;
-                            SURCHARGE_CACHE
-                                .remove(CacheKey {
-                                    key: key.to_string(),
-                                    prefix: self.key_prefix.clone(),
-                                })
-                                .await;
-
-                            key
-                        }
-                    };
-
-                    self.delete_key(key.as_ref())
-                        .await
-                        .map_err(|err| logger::error!("Error while deleting redis key: {err:?}"))
-                        .ok();
-
-                    logger::debug!(
-                        "Handled message on channel {channel_name} - Done invalidating {key}"
-                    );
+                cache::IMC_INVALIDATION_CHANNEL => {
+                    let keys_to_be_invalidated =
+                        match CacheKind::from_redis_value(RedisValue::new(message.value))
+                            .change_context(redis_errors::RedisError::OnMessageError)
+                        {
+                            Ok(value) => value,
+                            Err(err) => {
+                                logger::error!(value_conversion_err=?err);
```

Comment:
We can add a user-friendly message here, informing that we couldn't read/parse the message on the invalidation channel.